### ISSUE TITLE
Redundant column when using subtotal in columns 

### ISSUE THREAD

**Issue by shirdekel** [gdemin/expss#102]
I want to get subtotal of columns with overlap but doing so produces an extra column.

``` r
library(expss)
#> Loading required package: maditr
#> 
#> To modify variables or add new variables:
#>              let(mtcars, new_var = 42, new_var2 = new_var*hp) %>% head()
data(infert)

## extra 6-11yrs column
infert |>
  tab_cells(parity) |>
  tab_cols(
    total(),
    subtotal(
      education,
      "LESS THAN 12 Y.O." = levels(education)[1:2],
      "GREATER THAN 5 Y.O." = levels(education)[2:3]
    )
  ) |>
  tab_stat_cases() |>
  tab_pivot(
    stat_position = "inside_columns"
  )
```

<table class="gmisc_table" style="border-collapse: collapse; margin-top: 1em; margin-bottom: 1em;">
<thead>
<tr>
<th style="border-bottom: 1px solid grey; font-weight: 900; border-top: 2px solid grey; text-align: center;">
</th>
<th style="font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;">
 #Total 
</th>
<th style="font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;">
 0-5yrs 
</th>
<th style="font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;">
 6-11yrs 
</th>
<th style="font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;">
</th>
<th style="font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;">
 LESS THAN 12 Y.O. 
</th>
<th style="font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;">
 12+ yrs 
</th>
<th style="font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;">
 GREATER THAN 5 Y.O. 
</th>
</tr>
</thead>
<tbody>
<tr>
<td colspan="8" style="font-weight: 900;">
 parity 
</td>
</tr>
<tr>
<td style="text-align: left;">
   1 
</td>
<td style="text-align: right;">
99
</td>
<td style="text-align: right;">
3
</td>
<td style="text-align: right;">
42
</td>
<td style="text-align: right;">
42
</td>
<td style="text-align: right;">
45
</td>
<td style="text-align: right;">
54
</td>
<td style="text-align: right;">
96
</td>
</tr>
<tr>
<td style="text-align: left;">
   2 
</td>
<td style="text-align: right;">
81
</td>
<td style="text-align: right;">
</td>
<td style="text-align: right;">
42
</td>
<td style="text-align: right;">
42
</td>
<td style="text-align: right;">
42
</td>
<td style="text-align: right;">
39
</td>
<td style="text-align: right;">
81
</td>
</tr>
<tr>
<td style="text-align: left;">
   3 
</td>
<td style="text-align: right;">
36
</td>
<td style="text-align: right;">
</td>
<td style="text-align: right;">
21
</td>
<td style="text-align: right;">
21
</td>
<td style="text-align: right;">
21
</td>
<td style="text-align: right;">
15
</td>
<td style="text-align: right;">
36
</td>
</tr>
<tr>
<td style="text-align: left;">
   4 
</td>
<td style="text-align: right;">
18
</td>
<td style="text-align: right;">
3
</td>
<td style="text-align: right;">
12
</td>
<td style="text-align: right;">
12
</td>
<td style="text-align: right;">
15
</td>
<td style="text-align: right;">
3
</td>
<td style="text-align: right;">
15
</td>
</tr>
<tr>
<td style="text-align: left;">
   5 
</td>
<td style="text-align: right;">
6
</td>
<td style="text-align: right;">
</td>
<td style="text-align: right;">
3
</td>
<td style="text-align: right;">
3
</td>
<td style="text-align: right;">
3
</td>
<td style="text-align: right;">
3
</td>
<td style="text-align: right;">
6
</td>
</tr>
<tr>
<td style="text-align: left;">
   6 
</td>
<td style="text-align: right;">
8
</td>
<td style="text-align: right;">
6
</td>
<td style="text-align: right;">
</td>
<td style="text-align: right;">
</td>
<td style="text-align: right;">
6
</td>
<td style="text-align: right;">
2
</td>
<td style="text-align: right;">
2
</td>
</tr>
<tr>
<td style="border-bottom: 2px solid grey; text-align: left;">
   #Total cases 
</td>
<td style="border-bottom: 2px solid grey; text-align: right;">
248
</td>
<td style="border-bottom: 2px solid grey; text-align: right;">
12
</td>
<td style="border-bottom: 2px solid grey; text-align: right;">
120
</td>
<td style="border-bottom: 2px solid grey; text-align: right;">
120
</td>
<td style="border-bottom: 2px solid grey; text-align: right;">
132
</td>
<td style="border-bottom: 2px solid grey; text-align: right;">
116
</td>
<td style="border-bottom: 2px solid grey; text-align: right;">
236
</td>
</tr>
</tbody>
</table>

``` r

## no extra column
infert |>
  tab_cells(parity) |>
  tab_cols(
    total(),
    subtotal(
      education,
      "LESS THAN 12 Y.O." = levels(education)[1:2],
      "GREATER THAN 5 Y.O." = levels(education)[2:3]
    )
  ) |>
  tab_stat_cases() |>
  tab_pivot()
```

<table class="gmisc_table" style="border-collapse: collapse; margin-top: 1em; margin-bottom: 1em;">
<thead>
<tr>
<th style="border-bottom: 1px solid grey; font-weight: 900; border-top: 2px solid grey; text-align: center;">
</th>
<th style="font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;">
 #Total 
</th>
<th style="font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;">
 0-5yrs 
</th>
<th style="font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;">
 6-11yrs 
</th>
<th style="font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;">
 LESS THAN 12 Y.O. 
</th>
<th style="font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;">
 6-11yrs 
</th>
<th style="font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;">
 12+ yrs 
</th>
<th style="font-weight: 900; border-bottom: 1px solid grey; border-top: 2px solid grey; text-align: center;">
 GREATER THAN 5 Y.O. 
</th>
</tr>
</thead>
<tbody>
<tr>
<td colspan="8" style="font-weight: 900;">
 parity 
</td>
</tr>
<tr>
<td style="text-align: left;">
   1 
</td>
<td style="text-align: right;">
99
</td>
<td style="text-align: right;">
3
</td>
<td style="text-align: right;">
42
</td>
<td style="text-align: right;">
45
</td>
<td style="text-align: right;">
42
</td>
<td style="text-align: right;">
54
</td>
<td style="text-align: right;">
96
</td>
</tr>
<tr>
<td style="text-align: left;">
   2 
</td>
<td style="text-align: right;">
81
</td>
<td style="text-align: right;">
</td>
<td style="text-align: right;">
42
</td>
<td style="text-align: right;">
42
</td>
<td style="text-align: right;">
42
</td>
<td style="text-align: right;">
39
</td>
<td style="text-align: right;">
81
</td>
</tr>
<tr>
<td style="text-align: left;">
   3 
</td>
<td style="text-align: right;">
36
</td>
<td style="text-align: right;">
</td>
<td style="text-align: right;">
21
</td>
<td style="text-align: right;">
21
</td>
<td style="text-align: right;">
21
</td>
<td style="text-align: right;">
15
</td>
<td style="text-align: right;">
36
</td>
</tr>
<tr>
<td style="text-align: left;">
   4 
</td>
<td style="text-align: right;">
18
</td>
<td style="text-align: right;">
3
</td>
<td style="text-align: right;">
12
</td>
<td style="text-align: right;">
15
</td>
<td style="text-align: right;">
12
</td>
<td style="text-align: right;">
3
</td>
<td style="text-align: right;">
15
</td>
</tr>
<tr>
<td style="text-align: left;">
   5 
</td>
<td style="text-align: right;">
6
</td>
<td style="text-align: right;">
</td>
<td style="text-align: right;">
3
</td>
<td style="text-align: right;">
3
</td>
<td style="text-align: right;">
3
</td>
<td style="text-align: right;">
3
</td>
<td style="text-align: right;">
6
</td>
</tr>
<tr>
<td style="text-align: left;">
   6 
</td>
<td style="text-align: right;">
8
</td>
<td style="text-align: right;">
6
</td>
<td style="text-align: right;">
</td>
<td style="text-align: right;">
6
</td>
<td style="text-align: right;">
</td>
<td style="text-align: right;">
2
</td>
<td style="text-align: right;">
2
</td>
</tr>
<tr>
<td style="border-bottom: 2px solid grey; text-align: left;">
   #Total cases 
</td>
<td style="border-bottom: 2px solid grey; text-align: right;">
248
</td>
<td style="border-bottom: 2px solid grey; text-align: right;">
12
</td>
<td style="border-bottom: 2px solid grey; text-align: right;">
120
</td>
<td style="border-bottom: 2px solid grey; text-align: right;">
132
</td>
<td style="border-bottom: 2px solid grey; text-align: right;">
120
</td>
<td style="border-bottom: 2px solid grey; text-align: right;">
116
</td>
<td style="border-bottom: 2px solid grey; text-align: right;">
236
</td>
</tr>
</tbody>
</table>

<sup>Created on 2023-01-18 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.2.2 (2022-10-31)
#>  os       macOS Monterey 12.6
#>  system   aarch64, darwin20
#>  ui       X11
#>  language (EN)
#>  collate  en_AU.UTF-8
#>  ctype    en_AU.UTF-8
#>  tz       Australia/Sydney
#>  date     2023-01-18
#>  pandoc   2.19.2 @ /opt/homebrew/bin/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version date (UTC) lib source
#>  backports     1.4.1   2021-12-13 [1] CRAN (R 4.2.0)
#>  checkmate     2.1.0   2022-04-21 [1] CRAN (R 4.2.0)
#>  cli           3.6.0   2023-01-09 [1] CRAN (R 4.2.0)
#>  data.table    1.14.6  2022-11-16 [1] CRAN (R 4.2.0)
#>  digest        0.6.31  2022-12-11 [1] CRAN (R 4.2.0)
#>  evaluate      0.20    2023-01-17 [1] CRAN (R 4.2.2)
#>  expss       * 0.11.4  2023-01-18 [1] Github (gdemin/expss@7a790ba)
#>  fastmap       1.1.0   2021-01-25 [1] CRAN (R 4.2.0)
#>  fs            1.5.2   2021-12-08 [1] CRAN (R 4.2.0)
#>  glue          1.6.2   2022-02-24 [1] CRAN (R 4.2.0)
#>  highr         0.10    2022-12-22 [1] CRAN (R 4.2.0)
#>  htmlTable     2.4.1   2022-07-07 [1] CRAN (R 4.2.0)
#>  htmltools     0.5.4   2022-12-07 [1] CRAN (R 4.2.0)
#>  htmlwidgets   1.6.1   2023-01-07 [1] CRAN (R 4.2.0)
#>  knitr         1.41    2022-11-18 [1] CRAN (R 4.2.0)
#>  lifecycle     1.0.3   2022-10-07 [1] CRAN (R 4.2.0)
#>  maditr      * 0.8.3   2022-04-02 [1] CRAN (R 4.2.0)
#>  magrittr      2.0.3   2022-03-30 [1] CRAN (R 4.2.0)
#>  matrixStats   0.63.0  2022-11-18 [1] CRAN (R 4.2.0)
#>  purrr         1.0.1   2023-01-10 [1] CRAN (R 4.2.0)
#>  R.cache       0.16.0  2022-07-21 [1] CRAN (R 4.2.0)
#>  R.methodsS3   1.8.2   2022-06-13 [1] CRAN (R 4.2.0)
#>  R.oo          1.25.0  2022-06-12 [1] CRAN (R 4.2.0)
#>  R.utils       2.12.2  2022-11-11 [1] CRAN (R 4.2.0)
#>  reprex        2.0.2   2022-08-17 [1] CRAN (R 4.2.0)
#>  rlang         1.0.6   2022-09-24 [1] CRAN (R 4.2.0)
#>  rmarkdown     2.19    2022-12-15 [1] CRAN (R 4.2.0)
#>  rstudioapi    0.14    2022-08-22 [1] CRAN (R 4.2.0)
#>  sessioninfo   1.2.2   2021-12-06 [1] CRAN (R 4.2.0)
#>  stringi       1.7.12  2023-01-11 [1] CRAN (R 4.2.0)
#>  stringr       1.5.0   2022-12-02 [1] CRAN (R 4.2.0)
#>  styler        1.9.0   2023-01-15 [1] CRAN (R 4.2.0)
#>  vctrs         0.5.1   2022-11-16 [1] CRAN (R 4.2.0)
#>  withr         2.5.0   2022-03-03 [1] CRAN (R 4.2.0)
#>  xfun          0.36    2022-12-21 [1] CRAN (R 4.2.0)
#>  yaml          2.3.6   2022-10-18 [1] CRAN (R 4.2.0)
#> 
#>  [1] /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>

**Comment by gdemin** [comment]
It seems it's a bug. What exactly do you want to achieve with ` tab_pivot(stat_position = "inside_columns")` in the case of a single stat?

**Comment by shirdekel** [comment]
I actually have two stats in my case but just used one for the reprex.

**Comment by gdemin** [comment]
Currently I onle can suggets to remove duplicated columns:
```R
library(expss)

data(infert)

remove_duplicate_columns = function(tbl){
    tr_tbl = t(as.matrix(tbl))
    dupl = duplicated(tr_tbl)
    tbl[,!dupl]
}


## extra 6-11yrs column
infert |>
  tab_cells(parity) |>
  tab_cols(
    total(),
    subtotal(
      education,
      "LESS THAN 12 Y.O." = levels(education)[1:2],
      "GREATER THAN 5 Y.O." = levels(education)[2:3]
    )
  ) |>
  tab_stat_cases() |>
  tab_pivot(
    stat_position = "inside_columns"
  ) %>% 
  remove_duplicate_columns()
```

**Comment by gdemin** [comment]
Fixed in 0.11.6
